eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BitcoinCoreClient.scala

```diff
@@ -130,18 +132,24 @@ class BitcoinCoreClient(val rpcClient: BitcoinJsonRPCClient) extends OnChainWall
    * @param outputIndex   index of the transaction output that has been spent.
    * @return the transaction spending the given output.
    */
-  def lookForSpendingTx(blockhash_opt: Option[ByteVector32], txid: ByteVector32, outputIndex: Int)(implicit ec: ExecutionContext): Future[Transaction] =
+  def lookForSpendingTx(blockhash_opt: Option[ByteVector32], txid: ByteVector32, outputIndex: Int)(implicit ec: ExecutionContext): Future[Transaction] = {
```

Comment:
Can't we define the right wrapper types in `bitcoin-lib` to avoid juggling with kotlin types here?

Reply:
it's linked to the comment above: without scala wrappers for `Block` we end up dealing with java lists, and since implicit conversions to Scala lists are deprecated we need to use `asScala` explicitly.